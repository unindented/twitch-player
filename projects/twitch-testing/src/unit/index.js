import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import React from "react";
import { render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>{node}</ThemeProvider>,
    ...options
  );
};

const fireResizeEvent = (el, { left = 0, top = 0, width = 0, height = 0 }) => {
  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const evt = document.createEvent("Event");
  evt.initEvent("resize", true, true);
  el.dispatchEvent(evt);
};

export * from "react-testing-library";

export { customRender as render, fireResizeEvent };
