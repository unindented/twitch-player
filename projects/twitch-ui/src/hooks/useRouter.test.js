import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import { Text } from "react-native";
import { MemoryRouter as Router } from "react-router";
import { useRouter } from ".";

const SomeComponent = () => {
  const { location } = useRouter();

  return <Text>{location.pathname}</Text>;
};

describe("useRouter", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <Router>
        <SomeComponent />
      </Router>
    );
  });

  it("renders with the current location", () => {
    expect(instance.asFragment()).toMatchSnapshot();
  });
});
