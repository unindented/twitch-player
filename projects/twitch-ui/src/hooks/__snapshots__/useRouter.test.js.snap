// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRouter renders with the current location 1`] = `
<DocumentFragment>
  <div
    class="css-text-76zvg2"
    dir="auto"
  >
    /
  </div>
</DocumentFragment>
`;
